packages/flutter_tools/lib/src/macos/xcdevice.dart

```diff
@@ -363,7 +373,20 @@ class XCDevice {
       // "13.3 (17C54)"
       final RegExp operatingSystemRegex = RegExp(r'(.*) \(.*\)$');
       final String operatingSystemVersion = deviceProperties['operatingSystemVersion'] as String;
-      return operatingSystemRegex.firstMatch(operatingSystemVersion.trim())?.group(1);
+      if(operatingSystemRegex.hasMatch(operatingSystemVersion.trim())) {
+        return operatingSystemRegex.firstMatch(operatingSystemVersion.trim())?.group(1);
```

Comment:
i added a condition to check here if the regex had a match , because it was returning null for 10.1 . i think it should return 10.1.